git-cliff-core/src/changelog.rs

```diff
@@ -434,13 +433,44 @@ impl<'a> Changelog<'a> {
 				bitbucket_commits.clone(),
 				bitbucket_pull_request.clone(),
 			)?;
+		}
+
+		// render the header_template
+		if let Some(header_template) = &self.header_template {
+			let write_result = writeln!(
+				out,
+				"{}",
+				header_template.render(
+					&Releases {
+						releases: &releases,
+					},
+					Some(&additional_context),
+					&postprocessors,
+				)?
+			);
+			if let Err(e) = write_result {
+				if e.kind() != std::io::ErrorKind::BrokenPipe {
+					return Err(e.into());
+				}
+			}
+
+			// always insert an empty line after the header
+			if let Err(e) = out.write(b"\n") {
+				if e.kind() != std::io::ErrorKind::BrokenPipe {
+					return Err(e.into());
+				}
+			}
```

Comment:
Do we want to do this? I think it can be left to user to insert a new line.

Reply:
Without this there can never be an empty line after header unless  `changelog.trim` is `false`, which would also impact the body and footer. Given that the body should always start with a heading, I think the forced new line after the header is a reasonable choice. But I am not entirely happy with it.

Reply:
After some more testing I concluded that there is no general solution that preserves new-line handling in all tests. We either enforce a new-line after the header or not and either the rust tests in `changelog.rs` or the test fixtures break. I would opt for the former because it has less impact on users. Thus I removed this way of forcing a new-line.